src/autocomplete.js

```diff
@@ -165,7 +164,7 @@ export default class Autocomplete extends Component {
       menuOpen: newState.menuOpen || false,
       query: newQuery,
       selected: null
-    })
+    }, this.props.confirmOnBlur ? this.props.onConfirm(newQuery) : null)
```

Comment:
> `options[selected]` was weird given that `newQuery` exists in the same scope.

Actually, disregard what I said. It should be `options[selected]`. `options[-1]` returning `undefined` is correct behaviour. In which tests did you find this to be unexpected?